File: src/main/java/no/fdk/userapi/UserApiApplication.java
package no.fdk.userapi;

import no.fdk.userapi.configuration.HostProperties;
import no.fdk.userapi.configuration.SecurityProperties;
import no.fdk.userapi.configuration.WhitelistProperties;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.context.properties.EnableConfigurationProperties;

@SpringBootApplication
@EnableConfigurationProperties({ HostProperties.class, SecurityProperties.class, WhitelistProperties.class })
public class UserApiApplication {
    public static void main(String[] args) {
        SpringApplication.run(UserApiApplication.class, args);
    }
}

File: src/main/kotlin/no/fdk/userapi/UserApiApplication.kt
package no.fdk.userapi

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.ConfigurationPropertiesScan

@SpringBootApplication
@ConfigurationPropertiesScan
class UserApiApplication

fun main(args: Array<String>) {
    SpringApplication.run(UserApiApplication::class.java, *args)
}
